app/tests/test_routes.py

```diff
@@ -244,3 +246,94 @@ def test_route_generate_image_manifest(
 
         assert response.status_code == 200
         assert actual_manifest == expected_image_manifest
+
+    @pytest.mark.parametrize(
+        "user_role, form_data, args_data, expected_redirect_route, expected_params",
+        [
+            # standard user with both form and args data, args has precedence for overlapping keys
+            (
+                "standard_user",
+                {
+                    "transferring_body_id": "form_value",
+                    "other_param": "form_other_value",
+                },
+                {
+                    "transferring_body_id": "args_value",
+                    "another_param": "args_another_value",
+                },
+                "main.search_transferring_body",
+                {
+                    "_id": "args_value",
+                    "other_param": "form_other_value",
+                    "another_param": "args_another_value",
+                },
+            ),
+            # standard user with only form data, no transferring_body_id in args
+            (
+                "standard_user",
+                {"transferring_body_id": "form_value"},
+                {},
+                "main.search_transferring_body",
+                {"_id": "form_value"},
+            ),
+            # standard user with only args data, transferring_body_id present
+            (
+                "standard_user",
+                {},
+                {"transferring_body_id": "args_value"},
+                "main.search_transferring_body",
+                {"_id": "args_value"},
+            ),
+            # all access user with args data (redirect to search_results_summary)
+            (
+                "all_access_user",
+                {},
+                {"some_param": "some_value"},
+                "main.search_results_summary",
+                {"some_param": "some_value"},
+            ),
+            # all access user with form data and args data (args takes precedence)
+            (
+                "all_access_user",
+                {"some_param": "form_value"},
+                {"some_param": "args_value"},
+                "main.search_results_summary",
+                {"some_param": "args_value"},
+            ),
+        ],
+    )
+    def test_search_route_with_various_cases(
+        app,
+        client: FlaskClient,
+        user_role,
+        form_data,
+        args_data,
+        expected_redirect_route,
+        expected_params,
+        mock_standard_user,
+        mock_all_access_user,
+    ):
+        if user_role == "standard_user":
```

Comment:
can we split this test function into tests that test one scenario each. Having conditionals in the tests makes them harder to follow as documentation pieces and harder to debug due to all the diverging paths